File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/HouseholdSummaryDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.Instant;
import java.util.UUID;

public record HouseholdSummaryDto(
    UUID id,
    String name,
    String role,
    String status,
    Instant createdAt
) {
    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getRole() {
        return role;
    }

    public String getStatus() {
        return status;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/HouseholdSummaryDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.time.Instant
import java.util.UUID

data class HouseholdSummaryDto(
    val id: UUID,
    val name: String,
    val role: String,
    val status: String,
    val createdAt: Instant,
)
